File: src/main/java/arcana/utils/wrappers/ContainerWrapper.java
package arcana.utils.wrappers;

import net.minecraft.inventory.container.*;
import net.minecraft.util.text.ITextComponent;
import net.minecraft.util.text.StringTextComponent;
import net.minecraftforge.common.extensions.IForgeContainerType;
import net.minecraftforge.fml.network.IContainerFactory;

public class ContainerWrapper {
    public static <T extends Container> ContainerType<T> simple(String name, ContainerType.IFactory<T> factory){
        return (ContainerType<T>) new ContainerType<>(factory).setRegistryName(name);
    }

    public static <T extends Container> ContainerType<T> withExtraData(String name, IContainerFactory<T> factory){
        return (ContainerType<T>) new ContainerType<>(factory).setRegistryName(name);
    }

    private static final ITextComponent empty = new StringTextComponent("");
    public static INamedContainerProvider namelessProvider(IContainerProvider provider){
        return new SimpleNamedContainerProvider(provider, empty);
    }
}

File: src/main/kotlin/arcana/utils/wrappers/ContainerWrapper.kt
package arcana.utils.wrappers

import net.minecraft.inventory.container.*
import net.minecraft.util.text.ITextComponent
import net.minecraft.util.text.StringTextComponent
import net.minecraftforge.fml.network.IContainerFactory

object ContainerWrapper {
    fun <T : Container?> simple(name: String, factory: ContainerType.IFactory<T>): ContainerType<T> {
        return ContainerType(factory).setRegistryName(name) as ContainerType<T>
    }

    fun <T : Container?> withExtraData(name: String, factory: IContainerFactory<T>): ContainerType<T> {
        return ContainerType(factory).setRegistryName(name) as ContainerType<T>
    }

    private val empty: ITextComponent = StringTextComponent("")
    fun namelessProvider(provider: IContainerProvider): INamedContainerProvider {
        return SimpleNamedContainerProvider(provider, empty)
    }
}